rfc0010-network-discovery.md

```
@@ -0,0 +1,236 @@
+
+
+### Probing mechanism
+1. immediate 0-hop: observe only whether ACK arrived from the counterparty and how long it took for it to arrive.
+
+
+Algorithm:
+- discovery algorithm works in competing modes: bread and width first
+- basic operations:
+  1. discover immediate peers
+  2. generate paths for n-hop connections (referential probing = low frequency)
+  3. for sessions prepopulate the cache from 
```

markdownlint-cli2 (0.17.2)

Trailing spaces
Expected: 0 or 2; Actual: 1

(MD009, no-trailing-spaces)